### PR TITLE
Fix building with GCC 10

### LINKED ISSUE #1341: 4.4.3 Build fails with both GCC 10 and Clang 10
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
Building 4.4.3 fails for different reasons with GCC 10 and Clang 10. FYI I can "fix" the clang build by temporarily passing `-Wno-return-type`.

### Logs
GCC: https://buildbot.alucryd.xyz/#/builders/213/builds/10/steps/7/logs/stdio
Clang: https://buildbot.alucryd.xyz/#/builders/213/builds/11/steps/7/logs/stdio

## Platform Information
- [ ] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

Arch Linux, GCC 10, Clang 10.

### PR DESCRIPTION
**Pull request by jeremypw**
Attempt to fix #1341 (GCC part)

The symbol name "settings" is not used and is replaced by names specific to where it was used in order to try and avoid errors such as 
```
(.text+0x0): multiple definition of `settings'; src/25a6634@@io.elementary.files@exe/meson-generated_Application.c.o (symbol from plugin):(.text+0x0): first defined here
/usr/bin/ld: src/25a6634@@io.elementary.files@exe/meson-generated_View_Window.c.o (symbol from plugin): in function `settings':
```

when compiling with GCC 10.

"settings" is also removed from the Preferences namespace in `marlin.vapi` (This vapi may no longer be needed anyway as certain settings are exposed to the library through the GOF.Preferences object).

### CODE REVIEW

**Comment by jeremypw** [comment]
MarlinIconInfoTests failing for no apparent reason :-(

**Comment by jeremypw** [comment]
I can replicate the CI failure locally by uninstalling Files (with its schema) and then trying `ninja test`.  What is puzzling is that the tests have not changed and previous passed (and pass in master).

**Comment by jeremypw** [comment]
Seems the current commit is not sufficient to fix GCC10 compilation (compiles with GCC9.3) so need to make more changes without breaking CI.

**Comment by jeremypw** [comment]
With the latest commit, the PR both builds with GCC10 and passes CI.

File: src/Application.vala [hunk 1]
```diff
@@ -31,10 +31,25 @@ public class Marlin.Application : Gtk.Application {
     private const uint MAX_WINDOWS = 25;
 
     public int window_count { get; private set; }
+    public Settings marlin_app_settings { get; construct; }
+    public Settings marlin_icon_view_settings { get; construct; }
+    public Settings marlin_list_view_settings { get; construct; }
+    public Settings marlin_column_view_settings { get; construct; }
+    public Settings gnome_interface_settings { get; construct; }
+    public Settings gnome_privacy_settings { get; construct; }
+    public Settings gtk_file_chooser_settings { get; construct; }
```
Comment (danrabbit) [inline src/Application.vala:40]: Should these be constructed statically instead? Then you wouldn't have to reference the instance as in `slot.window.marlin_app.marlin_app_settings` which is kind of verbose

Comment (jeremypw) [reply src/Application.vala]: Done.

**Pull request merged**